src/Tribe/Events/Attendee_List_Display.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace Tribe\Tickets\Events;
+
+use Exception;
+use InvalidArgumentException;
+use Tribe__Tickets_Plus__Attendees_List;
+use WP_Post;
+
+class Attendee_List_Display {
+
+	private static $attendee_list_by_shortcode = 'tribe_tickets_attendee_list_triggered_by_shortcode';
+
+	/**
+	 * Check if given event is hiding the attendees list.
+	 *
+	 * @param int|WP_Post $post
+	 *
+	 * @throws InvalidArgumentException
+	 *
+	 * @return bool True if event is hiding the attendees list. False otherwise.
+	 */
+	public function is_event_hiding_attendee_list( $post ) {
+		// Attendees list is a Plus feature, if ET Plus is not present it will always be hidden.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return true;
+		}
+
+		if ( is_numeric( $post ) ) {
+			$post = WP_Post::get_instance( $post );
+		}
+
+		if ( ! $post instanceof WP_Post ) {
+			/*
+			 * There's no way to apply a sensible default in this scenario.
+			 *
+			 * Let's throw an exception so the client calling this function deals
+			 * with it according to the >> caller << context.
+			 */
+			throw new InvalidArgumentException( '$post should be either a post ID or a WP_Post object.' );
+		}
+
+		return (bool) Tribe__Tickets_Plus__Attendees_List::is_hidden_on( $post );
+	}
+
+	/**
+	 * @filter tribe_tickets_plus_hide_attendees_list_optout 10 1
+	 *
+	 * @see \Tribe\Tickets\Events\Events_Service_Provider::hooks
+	 *
+	 * @return bool
+	 */
+	public function should_hide_optout( $should_hide ) {
+		try {
+			global $post;
+
+			return $this->is_event_hiding_attendee_list( $post );
+		} catch ( Exception $e ) {
+			// Eg: global $post not a WP_Post object
+			return $should_hide;
+		}
+	}
+
+	/**
+	 * Determines whether this post is displaying the Attendees List.
+	 *
+	 * @action save_post_tribe_events 10 1
+	 *
+	 * @see \Tribe\Tickets\Events\Events_Service_Provider::hooks
+	 *
+	 * @retyrn void
+	 */
+	public function maybe_update_attendee_list_hide_meta( $post ) {
+		// Attendees list is a Plus feature, if ET Plus is not present we don't have to update the meta.
+		if ( ! class_exists( 'Tribe__Tickets_Plus__Main' ) ) {
+			return;
+		}
+
+		// don't do anything on autosave, auto-draft, or massupdates
+		if ( wp_is_post_autosave( $post ) || wp_is_post_revision( $post ) ) {
+			return;
+		}
+
+		if ( ! $post instanceof WP_Post ) {
+			$post = get_post( $post );
+		}
+
+		// Bail on Invalid post
+		if ( ! $post instanceof WP_Post ) {
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: cb73e624c7e407fb04fbca10befa8de0</sub>